### PR TITLE
More coverity fixes

### PR DESCRIPTION
**Pull request by acv**
Fixes 4 new warning I introduced and a bunch of other warnings.

Also removed most `extern` definitions. Too many C files knew too much about the innards of other files!

### CODE REVIEW

File: src/wdctl_thread.c [hunk 1]
```diff
@@ -290,7 +288,10 @@ wdctl_restart(int afd)
 	}
 
 	debug(LOG_DEBUG, "Creating socket");
-	sock = socket(PF_UNIX, SOCK_STREAM, 0);
+    if (-1 == (sock = socket(PF_UNIX, SOCK_STREAM, 0))) {
+        debug(LOG_ERR, "Could not created wdctl socket (%s)", strerror(errno));
+        exit(1);
```
Comment (mhaas) [inline src/wdctl_thread.c]: Didn't you want me to call terminate here? Or is that automatically called on exit()?

File: src/wdctl_thread.c [hunk 2]
```diff
@@ -114,16 +111,17 @@ thread_wdctl(void *arg)
...
 	
 	/* Which to use, AF_UNIX, PF_UNIX, AF_LOCAL, PF_LOCAL? */
-	if (bind(wdctl_socket_server, (struct sockaddr *)&sa_un, strlen(sock_name) 
-				+ sizeof(sa_un.sun_family))) {
+	if (-1 == bind(wdctl_socket_server, (struct sockaddr *)&sa_un, sizeof(struct sockaddr_un))) {
 		debug(LOG_ERR, "Could not bind control socket: %s",
 				strerror(errno));
+        close(wdctl_socket_server);
 		pthread_exit(NULL);
 	}
```
Comment (mhaas) [inline src/wdctl_thread.c]: Pthread_exit, exit or terminate?

File: src/wdctl_thread.c [hunk 3]
```diff
@@ -304,13 +305,15 @@ wdctl_restart(int afd)
...
 	
 	/* Which to use, AF_UNIX, PF_UNIX, AF_LOCAL, PF_LOCAL? */
-	if (bind(sock, (struct sockaddr *)&sa_un, strlen(sock_name) + sizeof(sa_un.sun_family))) {
+	if (-1 == bind(sock, (struct sockaddr *)&sa_un, sizeof(struct sockaddr_un))) {
 		debug(LOG_ERR, "Could not bind internal socket: %s", strerror(errno));
+        close(sock);
 		return;
 	}
 
 	if (listen(sock, 5)) {
```
Comment (mhaas) [inline src/wdctl_thread.c]: This check should be for -1?

Quoting the man page: 

On success, zero is returned. On error, -1 is returned, and errno is set appropriately.

Comment (acv) [inline src/wdctl_thread.c]: I like explicit, but technically, -1 == true in C. Or more correctly false == 0 && true == !false

File: src/wdctl_thread.c (hunk 1 above)
Comment (acv) [inline src/wdctl_thread.c]: I will change to terminate everywhere.

File: src/wdctl_thread.c (hunk 2 above)
Comment (acv) [inline src/wdctl_thread.c]: I will change to terminate all calls to pthread_exit() in this file.

File: src/wdctl_thread.c (hunk 3 above)
Comment (mhaas) [inline src/wdctl_thread.c]: Yeah, but I'd be good to have it explicit for the less gifted like me. I typically do Python and Java ;)

Comment (acv) [inline src/wdctl_thread.c]: I'll take a chainsaw to that file and refactor the network calls into discrete function and try to make it clearer. There's already too much duplication for my taste.

**Comment by acv** [comment]
Merged in @mhaas's fixes and some comments.

**Pull request merged**